src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5489,13 +5487,89 @@ private void BindMemberAccessReportError(BoundMethodGroup node, DiagnosticBag di
                 {
                     Error(diagnostics, ErrorCode.ERR_NoSuchMemberOrExtensionNeedUsing, name, boundLeft.Type, plainName, "System");
                 }
+                else if (boundLeft.Type.IsTupleType && IsMissingMemberAnInferredTupleName(boundLeft, plainName))
+                {
+                    Error(diagnostics, ErrorCode.ERR_TupleInferredNamesNotAvailable, name, boundLeft.Type, plainName,
+                        new CSharpRequiredLanguageVersion(LanguageVersion.CSharp7_1));
+                }
                 else
                 {
                     Error(diagnostics, ErrorCode.ERR_NoSuchMemberOrExtension, name, boundLeft.Type, plainName);
                 }
             }
         }
 
+        /// <summary>
+        /// When reporting a missing member error on a tuple type, determine whether that member
+        /// could be an element with an inferred name (had the language version been above 7.1).
+        /// </summary>
+        private bool IsMissingMemberAnInferredTupleName(BoundExpression boundLeft, string plainName)
+        {
+            Debug.Assert(boundLeft.Type.IsTupleType);
+            if (Compilation.LanguageVersion >= LanguageVersion.CSharp7_1)
+            {
+                return false;
+            }
+
+            var declarations = boundLeft.Type.DeclaringSyntaxReferences;
+            if (declarations.Count() != 1)
+            {
+                return false;
+            }
+
+            var syntax = declarations[0].GetSyntax();
+            if (syntax.Kind() != SyntaxKind.TupleExpression)
+            {
+                return false;
+            }
+
+            var arguments = ((TupleExpressionSyntax)syntax).Arguments;
+            var (elementNamesArray, inferredArray) = ExtractTupleElementNames(arguments);
+
+            for (int i = 0; i < arguments.Count; i++)
+            {
+                if (elementNamesArray[i]?.Equals(plainName, StringComparison.Ordinal) == true && inferredArray[i])
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private (ImmutableArray<string> elementNamesArray, ImmutableArray<bool> inferredArray) ExtractTupleElementNames(
+            SeparatedSyntaxList<ArgumentSyntax> arguments)
+        {
+            int numElements = arguments.Count;
+            var uniqueFieldNames = PooledHashSet<string>.GetInstance();
+            ArrayBuilder<string> elementNames = null;
```

Comment:
I followed the pre-existing pattern. It's probably premature optimization, but if there are not names, we won't event get an ArrayBuilder... In the `MergeTupleElementNamesAndFree` method, this results in using `default(ImmutableArray<string>)`.

I'd be ok to change. 
@VSadov what do you think?